### PR TITLE
Basic features for library

### PR DESCRIPTION
**Pull request by zero-nnkn**
## Description
- Add the basic features for the library: convert file (`*.heic`, `*.tiff`, `*.jpg`, `*.pdf`) to images (`*.png`).
- Add unit tests.

## Type of change
- [x] New features (non-breaking change).
- [ ] Bug fix (non-breaking change).
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected).

## Checklist
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [ ] There is a reference to the original bug report and related work.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] The feature is well documented.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pillow_heif==0.12.0
+pypdfium2==4.18.0
```
Comment (Tin-Hoang) [inline requirements.txt]: This repo is a library.
Hence, using `requirements.txt` with exact (pinned) package versions here would not be wise.
For a python library, we would prefer "abstract dependencies" because they should aim for redistributable things. The looser the packages' versions of a library are, the more likely that it would not cause dependency conflict on an application that utilizes multiple libraries.
For further reading, you can check this post:
https://caremad.io/posts/2013/07/setup-vs-requirement/
_tl;dr `requirements.txt` is suitable for a full-fledged application; Else, use `pyproject.toml` or `setup.py` file (legacy) if the repo is a library._

File: README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Pyimg-converter
+
+Convert file (`*.heic`, `*.tiff`, `*.jpg`, `*.pdf`) to images (`*.png`).
+
+## Installation
+```
+pip install -r requirements.txt
```
Comment (Tin-Hoang) [inline README.md]: As said in the above comment, 
A more proper way is to use `pyproject.toml` file from the repo template to install the library.
Hence, I suggest moving the required packages into `pyproject.toml`, and using loose versions of them.

----
Tip: An alternative command for installation, users can also directly install it with git+https instead of the trivial `git clone to local -> cd ... -> pip install ...`:
```
pip install git+https://github.com/zero-nnkn/pyimg-converter.git
```
Further read: https://pip.pypa.io/en/stable/topics/vcs-support/

File: src/pyimg_converter/convert.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      input_path (str): path of input file
+      output_path (str): specifies the path where the converted image(s) will be saved.
+    If `output_path` is not provided, the converted image(s) will not be saved to a file.
+
+    Returns:
+      A list of images.
+    """
+    input_path = Path(input_path)
+    assert input_path.exists(), f'Input path does not exist: {input_path}'
+    assert input_path.is_file(), f'Input path is not a file: {input_path}'
```
Comment (Tin-Hoang) [inline src/pyimg_converter/convert.py]: `assert` is convenient for debugging or unit-test code. But, it should be avoided in the production code.
A proper alternative is to catch the expectation and raise the correct corresponding code for that error.
Above code could be changed into:
```suggestion
    if not input_path.exists():
        raise FileNotFoundError(f"Input path does not exist: {input_path}")
    if not input_path.is_file():
        raise IsADirectoryError(f"Input path is not a file: {input_path}")
```

File: src/pyimg_converter/convert.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    file_type = input_path.suffix[1:].lower()
+    converter = get_converter(file_type)
+    images = converter.convert_from_path(input_path)
+
+    if output_path is not None:
+        assert output_path[-4:].lower() == '.png', 'Output path must be in *.png format'
```
Comment (Tin-Hoang) [inline src/pyimg_converter/convert.py]: Same as above,
We could change assert into a more informative error raise.

File: tests/unittests.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import PIL
+
+FILE = Path(__file__).resolve()
+ROOT = FILE.parents[1]
+if str(ROOT) not in sys.path:
+    sys.path.append(str(ROOT))
+ROOT = Path(os.path.relpath(ROOT, Path.cwd()))
+
+from src.pyimg_converter import convert_from_path
```
Comment (Tin-Hoang) [inline tests/unittests.py]: Nitpick: PEP 8: E402 module level import not at top of file
You can use `isort` to automate this.

**Review by Tin-Hoang — COMMENTED** [review]
Thanks, team,
Very neat library!
The use of the base abstraction class is clever.
I have a few suggestions below, pls check.

---

Besides, here are other things that you can make it better:
- Custom `pyproject.toml` for this project. (the authors, description is still from the original template)
- Add description for the PR. You can also think of [creating a GitHub template for Pull Request](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository#adding-a-pull-request-template).

**Comment by Tin-Hoang** [comment]
btw, kudos on the team collaboration.
I see that you are effectively following [trunk-based development for small team](https://trunkbaseddevelopment.com/#trunk-based-development-for-smaller-teams) for this short-lived feature branch.
image

File: src/pyimg_converter/convert.py (hunk 3 above)
Comment (zero-nnkn) [reply src/pyimg_converter/convert.py]: Thanks for your advice. I will update it using the error code.

File: src/pyimg_converter/convert.py (hunk 4 above)
Comment (zero-nnkn) [reply src/pyimg_converter/convert.py]: Thanks for your advice. I will update it using the error code.

File: requirements.txt (hunk 1 above)
Comment (nxquang-al) [reply requirements.txt]: We removed the `requirements.txt` and listed all the loose package versions in the `pyproject.toml`. We also shortened the `pyproject.toml`, please check it out. Thanks for your advice.

**Review by Tin-Hoang — APPROVED** [review]
